libsolidity/codegen/ArrayUtils.h

```diff
@@ -67,7 +70,7 @@ class ArrayUtils
 	/// Appends a loop that clears a sequence of storage slots of the given type (excluding end).
 	/// Stack pre: end_ref start_ref
 	/// Stack post: end_ref
-	void clearStorageLoop(Type const& _type) const;
+	void clearStorageLoop(TypePointer const& _type) const;
```

Comment:
The reference counters in the implementation of `shared_ptr` are `const` ed on this line.  Ah, I'm talking about the `TypePointer` being `const`.

I'm totally fine with `Type` being `const`, but `TypePointer` being `const` is weird.